webhooks/cluster.go

```diff
@@ -75,7 +77,22 @@ func (webhook *Cluster) Default(_ context.Context, obj runtime.Object) error {
 		if !strings.HasPrefix(cluster.Spec.Topology.Version, "v") {
 			cluster.Spec.Topology.Version = "v" + cluster.Spec.Topology.Version
 		}
+
+		var clusterClass clusterv1.ClusterClass
+		if err := webhook.Client.Get(ctx, client.ObjectKey{
+			Namespace: cluster.Namespace,
+			Name:      cluster.Spec.Topology.Class,
+		}, &clusterClass); err != nil {
+			return errors.Wrapf(err, "failed to get ClusterClass %s %s", cluster.Namespace, cluster.Spec.Topology.Class)
```

Comment:
Q: should this be apierrors.NewNotFound given that the error is going to bubble up via the apiserver?
same for validation